### PR TITLE
Bump object and thorin-dwp

### PR DESCRIPTION
**Pull request by workingjubilee**
Required to fix watchOS breakage.

### CODE REVIEW

**Comment by workingjubilee** [comment]
cc @vladimir-ea

File: Cargo.lock [hunk 1]
```diff
@@ -4621,7 +4648,7 @@ dependencies = [
  "hermit-abi 0.3.0",
  "libc",
  "miniz_oxide",
- "object",
```
Comment (xfix) [inline Cargo.lock:4624]: Is depending on old version of `object` for `std` intentional?

File: Cargo.lock [hunk 2]
```diff
@@ -4621,7 +4648,7 @@ dependencies = [
...
+ "object 0.30.3",
```
Comment (xfix) [inline Cargo.lock:4651]: Is depending on old version of `object` for `std` intentional?

Comment (workingjubilee) [reply Cargo.lock:4651]: Yes, it's pointless to bump this unless backtrace is updated first.

**Comment by klensy** [comment]
There also exist #111384(or maybe one more?) with same bump, so need to coordinate.

File: Cargo.lock [hunk 3]
```diff
@@ -2331,18 +2337,28 @@ dependencies = [
...
+version = "0.30.3"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "8d864c91689fdc196779b98dba0aceac6118594c2df6ee5d943eb6a8df4d107a"
+checksum = "ea86265d3d3dcb6a27fc51bd29a4bf387fae9d2986b823079d4986af253eb439"
 dependencies = [
  "compiler_builtins",
+ "memchr",
+ "rustc-std-workspace-alloc",
+ "rustc-std-workspace-core",
+]
```
Comment (klensy) [inline Cargo.lock:2348]: There also a minor bump of std one's object dependency.

Comment (workingjubilee) [reply Cargo.lock:2348]: Well, this doesn't matter because backtrace and std still agree.

**Comment by workingjubilee** [comment]
@petrochenkov Yes, that PR is now effectively a superset of this one on all points that matter.

File: Cargo.lock (hunk 3 above)
Comment (workingjubilee) [reply Cargo.lock]: I downgraded object to 0.30.1 anyways just to minimize later merge conflicts.

**Comment by Mark-Simulacrum** [comment]
r=me when ready

**Comment by imWildCat** [comment]
@Mark-Simulacrum sorry to jump into this thread.

No sure but #111384 seems to already contain changes in this PR.

**Comment by workingjubilee** [comment]
Is ready, yes.

@bors r=MarkSimulacrum

**Comment by workingjubilee** [comment]
@imWildCat It is the reviewer's intention to delay approving that PR until after this one is merged so we might as well get on with this one.

**Comment by imWildCat** [comment]
@workingjubilee understand. Thank you!

**Comment by bmisiak** [comment]
Excellent! I’ll rebase #111384  on top of this.

**Comment by workingjubilee** [comment]
I know no action was called for, but regarding the perf and 3.5 second bootstrap regression:

0.7 of these seconds is the object crate.

Some of this is inevitable complication. However, with the enabling of compression features in object, it is possible that either object or thorin-dwp are including (thus, compiling) more code than Rust actually uses now, due to the use of compression by default. This could be fixed if thorin-dwp allowed forwarding features so we could disable the compression. However, it is likely we want to instead make more deliberate use of that compression somehow.

**Comment by workingjubilee** [comment]
also: Is await-call-tree normally a bouncy benchmark? The wall-time massively improved on that.

**Pull request merged**